src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift

```diff
@@ -0,0 +1,405 @@
+		// GIVEN
+		let response: [String: String] = ["errorCode": "DEVICE_TOKEN_SYNTAX_ERROR"]
+		let stack = MockNetworkStack(
+			httpStatus: 403,
+			responseData: try JSONEncoder().encode(response)
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .deviceTokenSyntaxError)
+			}
+		}
+	}
+
+	func testGIVEN_AuthorizeOTP_WHEN_Failure_OtherServerErrorIsCalled_THEN_otherServerErrorIsReturned() throws {
+		// GIVEN
+		let response: [String: String] = ["errorCode": "JWS_SIGNATURE_VERIFICATION_FAILED"]
+		let stack = MockNetworkStack(
+			httpStatus: 403,
+			responseData: try JSONEncoder().encode(response)
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .otherServerError)
+			}
+		}
+	}
+
+	func testGIVEN_AuthorizeOTP_WHEN_Failure_500StatusCode_THEN_internalServerErrorIsReturned() throws {
+		// GIVEN
+		let response: [String: String] = ["errorCode": "JWS_SIGNATURE_VERIFICATION_FAILED"]
+		let stack = MockNetworkStack(
+			httpStatus: 500,
+			responseData: try JSONEncoder().encode(response)
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .otherServerError)
```

Comment:
the name of the test suggest that it should return internalServerError but in the end we are equating to otherServerError, both of them are different errors if we look into the enum types, am I missing something here?